scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,378 @@
+                    filesRelativePathName = recursiveFolderDownload(channelsftp,
+                                                                    sftpRemoteRelativePath,
+                                                                    sftpLocalRelativePath);
+                    getOut().println("END DOWNLOAD FROM SFTP");
+                    break;
+
+                //ftp mode is put
+                case "PUT":
+                    getOut().println("BEGIN UPLOAD TO SFTP SERVER: " + sftpUrlKey);
+                    // Create remote directory(ies) if it(they) does(do) not exist
+                    if (!remoteDirExists(channelsftp, sftpRemoteRelativePath)) {
+                        makeRemoteDirectories(channelsftp, Paths.get(sftpRemoteRelativePath));
+                    }
+                    // recursive folder content upload to sftp server
+                    filesRelativePathName = recursiveFolderUpload(channelsftp,
+                                                                  sftpLocalRelativePath,
+                                                                  sftpRemoteRelativePath);
+                    getOut().println("END UPLOAD TO SFTP");
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("SFTP mode can only be PUT or GET.");
+            }
+        } catch (JSchException e) {
+            throw new JSchException("An error occured while trying to connect to SFTP server: ", e);
+        } catch (SftpException e) {
+            throw new SftpException(e.id,
+                                    "File not found, permission denied or operation not supported by SFTP server.",
+                                    e);
+        } finally {
+            if (channelsftp != null)
+                channelsftp.disconnect();
+            if (channel != null)
+                channel.disconnect();
+            if (session != null)
+                session.disconnect();
+            getOut().println("Disconnected from SFTP server");
+        }
+
+        return (Serializable) filesRelativePathName;
+    }
+
+    /**
+     * This method recursively uploads a local file/folder to an SFTP server
+     *
+     * @param sourcePath
+     * @param destinationPath
+     * @throws SftpException
+     * @throws FileNotFoundException
+     */
+    private List<String> recursiveFolderUpload(ChannelSftp channelSftp, String sourcePath, String destinationPath)
+            throws SftpException, FileNotFoundException {
+
+        List<String> filesRelativePathName = new ArrayList<>();
+        File sourceFile = new File(sourcePath);
+        String remoteFilePath = Paths.get(destinationPath, sourceFile.getName()).toString();
+
+        if (sourceFile.isFile()) {
+            // copy if it is a file
+            getOut().println("Uploading " + sourcePath);
+            channelSftp.put(new FileInputStream(sourceFile), remoteFilePath, ChannelSftp.OVERWRITE);
```

Comment:
Well, we thought of this, but then we decided to do this later because we have other connectors and we want to homogenize some common parameters including this one.
The idea is to have a boolean `Overwrite` variables in all connectors that fit.
So it's a do-it-later one.